src/context.rs

```diff
@@ -1274,9 +1274,75 @@ impl Context {
 
     /// Extends a PCR with the specified digests.
     ///
+    /// # Arguments
+    /// *`pcr_handle`- A [PcrHandle] to the PCR slot that is to be extended.
+    /// *`digests` - The [DigestValues] with which the slot shall be extended.
+    ///
     /// # Details
     /// This method is used to cause an update to the indicated PCR. The digests param
     /// contains the digests for specific algorithms that are to be used.
+    ///
+    /// # Example
+    ///
+    /// ```rust, no_run
+    ///use tss_esapi::{
+    ///     Context, Tcti,
+    ///     constants::{
+    ///         algorithm::{HashingAlgorithm, Cipher},
+    ///         types::session::SessionType,
+    ///     },
+    ///     structures::{PcrSlot, DigestValues, Digest},
+    ///     session::SessionAttributesBuilder,
+    ///     handles::PcrHandle,
+    /// };
+    /// use std::convert::TryFrom;
+    ///
+    /// // Create context that uses Device TCTI.
+    /// let mut context = unsafe {
+    ///     Context::new(Tcti::Device(Default::default())).expect("Failed to create Context")
+    /// };
+    ///
+    /// // Create session for a pcr
+    /// let pcr_session = context
+    ///     .start_auth_session(
+    ///         None,
+    ///         None,
+    ///         None,
+    ///         SessionType::Hmac,
+    ///         Cipher::aes_256_cfb(),
+    ///         HashingAlgorithm::Sha256,
+    ///     )
+    ///     .expect("Failed to create session")
+    ///     .expect("Recived invalid handle");
+    /// let (session_attributes, session_attributes_mask) = SessionAttributesBuilder::new()
+    ///     .with_decrypt(true)
+    ///     .with_encrypt(true)
+    ///     .build();
+    /// context.tr_sess_set_attributes(pcr_session, session_attributes, session_attributes_mask)
+    ///     .expect("Failed to set attributes on session");
```

Comment:
I think these sections could be made invisible in the documentation, as they're not directly relevant to how to use the method

Reply:
I am not sure how to do that. Because I like the idea that the documentation needs to be able to build otherwise we get an error.

Reply:
You just need to prefix the lines with `#`, as described [here](https://doc.rust-lang.org/rustdoc/documentation-tests.html#hiding-portions-of-the-example). Something like:
```rust
///#use tss_esapi::{
///#     Context, Tcti,
///#     constants::{
...
```

Reply:
That is really neat. I wonder if we should actually make the code runnable with tests as well and then just show the good parts? Because in that case would could move all the "basic" context tests into the documentation and only show the good parts?

Reply:
Have look at the ```pcr_extend``` method. If you think that looks ok. Then I will change the rest of the methods to look like that. I must say that it feels like I am lying when I do not show everything that needs to be done in order to get things to work hehe.

Reply:
> That is really neat. I wonder if we should actually make the code runnable with tests as well and then just show the good parts? Because in that case would could move all the "basic" context tests into the documentation and only show the good parts?

That might be ok, though I do remember facing some issues when trying to do something like that because the doc tests were run in a weird way. Also, I think moving all Context tests might be complete overkill - maybe just the ones that test each method works well?

Reply:
Nah not remove everything from context test. Just like the basic. Like when test that calling the function doesn't result in an error.